Spec/Auth.swift

```diff
@@ -1735,13 +1735,22 @@ class Auth : QuickSpec {
             // RSA8b
             context("should support all TokenParams") {
 
-                let options = AblyTests.commonAppSetup()
                 let currentClientId = "client_string"
-                options.clientId = currentClientId
 
-                let rest = ARTRest(options: options)
+                var options: ARTClientOptions!
+                var rest: ARTRest!
+
+                func setupDependencies() {
+                    if (options == nil) {
+                        options = AblyTests.commonAppSetup()
+                        options.clientId = currentClientId
+                        rest = ARTRest(options: options)
+                    }
+                }
 
                 it("using defaults") {
+                    setupDependencies()
```

Comment:
Yes, there is. You just need to create a `beforeEach { }` inside the `context` but, in that particular case, I don't want to setup to every `it`.

Reply:
ok, thanks for explaining.

🤔 
_It could be a sign that we need to be splitting these out in to groups so that this kind of base functionality can be used. Always happy to see a refactor like that if it results in clarity - i.e. code that operates using standard mechanisms wherever possible as generally that'll be more maintainable going forwards._